app/backend/hiStoryBackend/api/views/auth/views_v1.py

```diff
@@ -1,12 +1,38 @@
 from django.http import JsonResponse
 from api.views.helpers import json_response_helper as jrh
-
+from django.contrib.auth.hashers import check_password, make_password
+import json
+from api.models import User
+import jwt
+from hiStoryBackend.settings import SECRET_KEY
 
 def sign_in(request):
-    if not request.method == 'POST':
-        return jrh.bad_request(['Unsupported HTTP verb'])
+	if not request.method == 'POST':
+		return jrh.bad_request(['Unsupported HTTP verb'])
+
+	content = json.loads(request.body)
+	if 'username' in content:
+		try:
+			user = User.objects.get(username = content['username'])
+		except User.DoesNotExist:
+			return JsonResponse({"errors": ["username does not exist"]})
+	elif 'email' in content:
+		try:
+			user = User.objects.get(email = content['email'])
+		except User.DoesNotExist:
+			return JsonResponse({"errors": ["email does not exist"]})
+	else:
+		return JsonResponse({"errors": ["username or email fields could not be found"]})
+
+	if check_password(content['password'], user.password_hash) == False:
+		return JsonResponse({"errors": ["wrong password"]})
+
+	if user.banned:
+		return JsonResponse({"errors": ["user is banned"]})
```

Comment:
This is **not** an error. Frontend should render a message saying that the user is banned from the system. This check is redundant.